block/manager.go

```diff
@@ -827,35 +827,47 @@ func (m *Manager) recordMetrics(block *types.Block) {
 }
 
 func (m *Manager) submitBlocksToDA(ctx context.Context) error {
-	submitted := false
+	submittedAllBlocks := false
 	backoff := initialBackoff
-	blocks := m.pendingBlocks.getPendingBlocks()
-	for attempt := 1; ctx.Err() == nil && !submitted && attempt <= maxSubmitAttempts; attempt++ {
-		res := m.dalc.SubmitBlocks(ctx, blocks)
+	blocksToSubmit := m.pendingBlocks.getPendingBlocks()
+	numTotalBlocks := len(blocksToSubmit)
```

Comment:
[nit] you can remove `numTotalBlocks` and directly use `len(blocksToSubmit)`

Reply:
I don't think you can since we set `blocksToSubmit = notSubmittedBlocks`.